quickfixj-core/src/main/java/quickfix/field/converter/UtcTimestampConverter.java

```diff
@@ -19,24 +19,33 @@
 
 package quickfix.field.converter;
 
+import org.quickfixj.SimpleCache;
 import quickfix.FieldConvertError;
 import quickfix.SystemTime;
 
 import java.text.DateFormat;
 import java.util.Calendar;
 import java.util.Date;
 import java.util.GregorianCalendar;
-import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * Convert between a timestamp and a String. A timestamp includes both a date
  * and a time.
  */
 public class UtcTimestampConverter extends AbstractDateTimeConverter {
     private static final ThreadLocal<UtcTimestampConverter> utcTimestampConverter = new ThreadLocal<>();
+    private final static SimpleCache<String, Long> dateCache = new SimpleCache<>(dateString -> {
+        final Calendar c = new GregorianCalendar(1970, 0, 1, 0, 0, 0);
+        c.setTimeZone(SystemTime.UTC_TIMEZONE);
+        final int year = Integer.parseInt(dateString.substring(0, 4));
+        final int month = Integer.parseInt(dateString.substring(4, 6));
+        final int day = Integer.parseInt(dateString.substring(6, 8));
+        c.set(year, month - 1, day);
+        return c.getTimeInMillis();
+    });
+
```

Comment:
@chrjohn this is an interesting function, I think we don't need a calendar for that, if it could be migrated to JDK 8 so avoid such invocation.

Also, date & time formatters now won't need `ThreadLocal`, I'll see if I can add more of that stuff on Friday, busy days ahead on my other project.